internal/metrics/metrics.go

```diff
@@ -1,17 +1,224 @@
+	pm.CertificatesExpiringIn7Days.Set(float64(expiringIn7DaysCertCount))
+	pm.CertificatesExpiringIn30Days.Set(float64(expiringIn30DaysCertCount))
+	pm.CertificatesExpiringIn90Days.Set(float64(expiringIn90DaysCertCount))
+}
+
+func certificateRequestsMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificate_requests",
+		Help: "Total number of certificate requests",
+	})
+	return metric
+}
+
+func outstandingCertificateRequestsMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "outstanding_certificate_requests",
+		Help: "Number of outstanding certificate requests",
+	})
+	return metric
+}
+
+func certificatesMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificates",
+		Help: "Total number of certificates provided to certificate requests",
+	})
+	return metric
+}
+
+func expiredCertificatesMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificates_expired",
+		Help: "Number of expired certificates",
+	})
+	return metric
+}
+
+func certificatesExpiringIn1DayMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificates_expiring_in_1_day",
+		Help: "Number of certificates expiring in less than 1 day",
+	})
+	return metric
+}
+
+func certificatesExpiringIn7DaysMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificates_expiring_in_7_days",
+		Help: "Number of certificates expiring in less than 7 days",
+	})
+	return metric
+}
+func certificatesExpiringIn30DaysMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificates_expiring_in_30_days",
+		Help: "Number of certificates expiring in less than 30 days",
+	})
+	return metric
+}
+
+func certificatesExpiringIn90DaysMetric() prometheus.Gauge {
+	metric := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "certificates_expiring_in_90_days",
+		Help: "Number of certificates expiring in less than 90 days",
+	})
+	return metric
+}
+
+func requestsTotalMetric() prometheus.CounterVec {
+	metric := prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_requests_total",
+			Help: "Tracks the number of HTTP requests.",
+		}, []string{"method", "code"},
+	)
+	return *metric
+}
+
+func requestDurationMetric() prometheus.HistogramVec {
+	metric := prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "http_request_duration_seconds",
+			Help:    "Tracks the latencies for HTTP requests.",
+			Buckets: prometheus.ExponentialBuckets(0.1, 1.5, 5),
+		}, []string{"method", "code"},
+	)
+	return *metric
+}
+
+func certificateExpiryDate(certString string) time.Time {
+	certBlock, _ := pem.Decode([]byte(certString))
+	cert, _ := x509.ParseCertificate(certBlock.Bytes)
+	// TODO: cert.NotAfter can exist in a wrong cert. We should catch that at the db level validation
```

Comment:
There are a bunch of TODO's in this PR, how are we tracking those?

Reply:
I can make a JIRA ticket for them for the future, I don't think it's relevant to this PR but it should not be ignored.

Reply:
Ok, please do

Reply:
TLSENG-243 created